@here/olp-sdk-core/lib/utils/multipartupload-internal/BlobV1UploadRequest.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { DataStoreRequestBuilder } from "@here/olp-sdk-core";
+import { BlobApi } from "@here/olp-sdk-dataservice-api";
+import { UploadRequest } from "./UploadRequest";
+
+/**
+ * @internal
+ * Implementation of `UploadRequest` for the Blob V1.
+ */
+export class BlobV1UploadRequest implements UploadRequest {
+    constructor(private readonly requestBuilder: DataStoreRequestBuilder) {}
+
+    async startMultipartUpload(opts: {
+        layerId: string;
+        handle: string;
+        contentType: string;
+        contentEncoding?: BlobApi.ContentEncodingEnum;
+        billingTag?: string;
+    }): Promise<{
+        uploadPartUrl: string;
+    }> {
+        const result = await BlobApi.startMultipartUpload(this.requestBuilder, {
+            dataHandle: opts.handle,
+            layerId: opts.layerId,
+            billingTag: opts.billingTag,
+            body: {
+                contentType: opts.contentType,
+                contentEncoding: opts.contentEncoding
+            }
+        });
+
+        if (
+            !result.links ||
+            !result.links.uploadPart ||
+            !result.links.uploadPart.href
+        ) {
+            return Promise.reject(
+                new Error(
+                    "Failed to start the multipart upload to Blob V1. Bad response."
+                )
+            );
+        }

```

Comment:
Is this it? Looking at the REST API at least 4 different HTTP error, besides the usual retriable 429 & 5xx, can be returned by the service: 400, 401, 403, 404. Don't you forward those to the user?

Reply:
in case of http errors we will not get this error, user will get HttpError as usual. It's just a checking if the response object is as expected